Plugins/Resources/Resources.cpp

```diff
@@ -0,0 +1,282 @@
+}
+
+int32_t Resources::OnGetList(CResGFF *pGff, CResList *pList, CResStruct *pStructure, char *szFieldID)
+{
+    if (strncmp(pGff->m_pFileType, "IFO", 3) == 0 && strcmp(szFieldID, "Mod_HakList") == 0)
+    {
+        return false;
+    }
+
+    return s_GetListHook->CallOriginal<int32_t>(pGff, pList, pStructure, szFieldID);
+}
+
+uint32_t Resources::LoadNwSyncResources()
+{
+    auto host = Config::Get<std::string>("NWSYNC_HOST");
+    auto manifest = Config::Get<std::string>("NWSYNC_MANIFEST");
+
+    if (!host || !manifest)
+    {
+        LOG_WARNING("Missing NWSync host/manifest variables. Skipping sync.");
+        return 0;
+    }
+
+    LOG_INFO("Loading NWSync manifest '%s/%s'", host.value(), manifest.value());
+
+    const auto &hashString = CExoString(manifest.value());
+    const auto &hash = Hash::SHA1{hashString};
+
+    NWSync::CNWCSync cli(host->c_str(), hash, "", "");
+    auto &status = cli.GetStatus();
+
+    while (true)
+    {
+        HttpConnection::UpdateAll();
+        cli.Update();
+
+        if (status.m_currentState == NWSYNC_STATE_WAIT_PERMISSION_TO_SYNC)
+        {
+            cli.SetSyncPermission(true);
+            continue;
+        }
+
+        ShowNWSyncProgressUpdate(status, false);
+
+        if (status.m_currentState >= NWSYNC_STATE_DONE)
+        {
+            break;
+        }
+
+        if (status.m_currentState == NWSYNC_STATE_SYNC_IN_PROGRESS)
+        {
+            // libcurl is happy to do the heavy lifting async
+            usleep(250);
+        }
+    }
+
+    if (status.m_currentState != NWSYNC_STATE_DONE)
+    {
+        LOG_ERROR("Failed to load NWSync manifest '%s/%s'", host.value(), manifest.value());
+        return LOADMODULE_ERROR_COULDNOTLOADHAK;
+    }
+
+    Globals::ExoResMan()->AddManifest(hash, RESMAN_PRIORITY_NWSYNC);
+    Globals::Rules()->ReloadAll();
+
+    return 0;
+}
+
+uint32_t Resources::LoadCustomHakResources()
+{
+    auto hakListStr = Config::Get<std::string>("CUSTOM_HAK_LIST", "");
+    std::vector<std::string> hakList = String::Split(hakListStr, ',');
+
+    if (!hakList.empty())
+    {
+        LOG_INFO("Using custom hak list: %s", hakListStr);
+    }
+    else
+    {
+        LOG_INFO("Skipping load of all hak files");
+    }
+
+    for (const std::string &hakName: hakList)
+    {
+        if (Globals::ExoResMan()->AddEncapsulatedResourceFile(CExoString("HAK:") + hakName, RESMAN_PRIORITY_USER_HAK))
+        {
+            LOG_INFO("Loaded hak '%s'", hakName);
+        }
+        else
+        {
+            LOG_ERROR("Failed to load hak '%s'", hakName);
+            return LOADMODULE_ERROR_COULDNOTLOADHAK;
+        }
+    }
+
+    return 0;
+}
+
+CExoString* Resources::FindNwSyncTlk()
+{
+    if (auto *pList = Globals::ExoResMan()->GetResOfType(Constants::ResRefType::TLK, false))
+    {
+        if (pList->m_nCount > 0)
+        {
+            // We can only load 1 custom tlk, so just use the first one we find.
+            return pList->m_pStrings[0];
```

Comment:
This isn't quite right. Ignoring the fact that someone could arbitrarily put many tlks in the manifest for no good reason, there is also a perfectly good reason to put both `custom.tlk` and `dialog.tlk` in there. So at very least, this should skip `dialog{,f}.tlk`